frontend/src/page/Realm.tsx

```diff
@@ -1,43 +1,65 @@
 import React from "react";
 import { Link } from "react-router-dom";
 
+import { graphql, useLazyLoadQuery } from "react-relay/hooks";
+import { RealmQuery } from "../query-types/RealmQuery.graphql";
+
 import { Breadcrumbs } from "../ui/Breadcrumbs";
 
 
 type Props = {
-    path: string[];
+    path: string;
 };
 
 export const Realm: React.FC<Props> = ({ path }) => {
-    const isRoot = path.length === 0;
-    const ids = resolvePath(path);
-    if (ids == null) {
+    const isRoot = path === "";
+
+    // TODO Build this query from fragments!
+    const { realm } = useLazyLoadQuery<RealmQuery>(graphql`
+        query RealmQuery($path: String!) {
+            realm: realmByPath(path: $path) {
+                name
+                parents {
+                    name
+                    path
+                }
+                children {
+                    id
+                    name
+                    path
+                }
```

Comment:
```suggestion
                parents { name path }
                children { id name path }
```

This looks better to me ... (even if this would require commas).

Reply:
No comma needed, and I don't care. :)

Reply:
Then I prefer collapsing those. Whitespace ain't cheap!!1